File: src/main/java/com/hjj/apiserver/common/ApiUtils.java
package com.hjj.apiserver.common;

public class ApiUtils {
    public static <T> ApiResponse_Java<T> success(T response) {
        return new ApiResponse_Java<>(true, response, null);
    }


    public static <T> ApiResponse_Java<T> success() {
        return new ApiResponse_Java<>(true, null, null);
    }

    public static ApiResponse_Java<?> error(String message, ApiError_Java.ErrCode errCode){
        return new ApiResponse_Java<>(false, null, new ApiError_Java(message, errCode));
    }
}

File: src/main/kotlin/com/hjj/apiserver/util/ApiUtils.kt
package com.hjj.apiserver.util

import com.hjj.apiserver.common.ApiError
import com.hjj.apiserver.common.ErrCode

class ApiUtils {
    companion object {
        fun <T> success(response: T):ApiResponse<T>{
            return ApiResponse(true, response, null)
        }
        fun success():ApiResponse<*>{
            return ApiResponse(true, null, null)
        }

        fun error(errorCode: ErrCode):ApiResponse<*>{
            return ApiResponse(false, null, ApiError(errorCode))
        }
    }
}